app/src/main/java/ar/com/wolox/android/cookbook/scanqr/ScanQrActivity.kt

```diff
@@ -0,0 +1,60 @@
+package ar.com.wolox.android.cookbook.scanqr
+
+import android.content.Intent
+import android.util.Log
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.scanqr.captureqr.CaptureQrActivity
+import ar.com.wolox.android.cookbook.scanqr.scanerror.ScanErrorFragment
+import ar.com.wolox.android.cookbook.scanqr.scanmenu.ScanMenuFragment
+import ar.com.wolox.android.cookbook.scanqr.scansuccess.ScanSuccessFragment
+import ar.com.wolox.wolmo.core.activity.WolmoActivity
+import com.google.zxing.integration.android.IntentIntegrator
+import javax.inject.Inject
+
+class ScanQrActivity : WolmoActivity(), ScanQrView {
+
+    @Inject
+    lateinit var presenter: ScanQrPresenter
+
+    override fun layout(): Int = R.layout.activity_scan_qr
+
+    override fun init() {
+        presenter.attachView(this)
+        showScanMenuFragment()
```

Comment:
Also, this should be on the presenter. The view shouldn't know what to do.

Reply:
you are right I forgot to move that